src/metabase/search/core.clj

```diff
@@ -122,11 +122,9 @@
    1 ; pool size
    1 ; max pool size
    0 TimeUnit/MILLISECONDS ; keepalive
-   (ArrayBlockingQueue. 1)
-   (ThreadPoolExecutor$DiscardPolicy.)))
```

Comment:
Repeating comment, because not sure the one I left on the diff will be seen.

Why remove the discard policy? The point is that is N processes trigger a reindex while another one is running, they'll all be satisfied by the next one. It doesn't make sense to throw an error!